lib/src/layer/tile_layer/tile_coordinates.dart

```diff
@@ -3,25 +3,32 @@ import 'dart:math';
 import 'package:meta/meta.dart';
 
 @immutable
-class TileCoordinates extends Point<int> {
+class TileCoordinates {
```

Comment:
Is there a reason we're not extending `Point` anymore? This is a breaking change (along with changing the constructor to be non-`const`), and I'm not sure what the motivation is.

Reply:
(The tests and lints are failing because of this change.)

Reply:
Right now only the tests should be broken. Happy to fix those up if we can find agreement.

> Is there a reason we're not extending Point anymore? 

Moistly two:
 * Compositionally, to me it's a code-smell. It's the old composition vs inheritance discussion. TileCoordinates "are" semantically not a specialization/extension of a 2D point. Even if you were to think of it more as a "mixin" relationship, the only method reused is scaleBy and even Point.distanceTo was ill-suited. Which is just another indication, that they're not parent/child but rather false friends.
 * (It's not super critical for this use-case but I know that I always get you on the performance front and I really dislike math.Point :) ) Dart implements generics as so-called reified generics. This means unlike language like C++ that unroll all their generic instantiations into specialized code, Dart (go, kotlin, ...) only emit code once that treats Point<int> and Point<double> as Point<num> internally, with the "reified" types only there to enforce type-safety along the edges.

A teenie benchmark to make my point:

```dart
class Point<T extends num>  {
  T x;
  T y;

  Point(this.x, this.y);

  Point<T> operator+(Point<T> other) {
    x = (x + other.x) as T;
    y = (y + other.y) as T;
    return this;
  }
}

class PointInt  {
  int x;
  int y;

  PointInt(this.x, this.y);

  PointInt operator+(PointInt other) {
    x += other.x;
    y += other.y;
    return this;
  }
}


void main() {
  const N = 1000000000;
  final funcs = {
    'PointInt': () {
      var x = PointInt(23, 23);
      var y = PointInt(42, 42);
      for (int i = 0; i < N; ++i) {
        x+= y;
      }
    },
    'Point<int>': () {
      var x = Point<int>(23, 23);
      var y = Point<int>(42, 42);
      for (int i = 0; i < N; ++i) {
        x+= y;
      }
    },
    // Needed, otherwise the compiler strips the generic.
    'Point<double>': () {
      var x = Point<double>(23, 23);
      var y = Point<double>(42, 42);
      for (int i = 0; i < N; ++i) {
        x+= y;
      }
    },
  };

  for (final MapEntry(key: k, value:  f) in funcs.entries) {
    final w = Stopwatch()..start();
    f();
    print('${k}: ${w.elapsed}');
  }
}
```

Make sure to compile to avoid any JIT warmup:

```
$ dart compile exe generic2.dart && ./generic2.exe                                                                                                          

PointInt: 0:00:00.285020
Point<int>: 0:00:04.055126
Point<double>: 0:00:07.007694
```

Which is a 14x difference because you're paying the virtual function overhead for each addition (which is literally the cheapest operation in itself).

I've always felt that for something as low-level as Point, implementing it as a generic was a poor choice just to save a little bit of code and binary size.

To finish my tangent, I'm sure FlutterMap would benefit from using a PointInt a PointDouble, it's just hard to quantify w/o putting a large amount of upfront work in first.

More importantly for this use-case though, I'd still challenge that TileCoordinates is a Point or benefits from being a point.

Reply:
Wow, I wasn't expecting that much of a performance improvement just be not using generics! You've definitely got me there.

My biggest issue is that this change is very breaking. We should at least try to implement feature parity with `Point`, ie. the operators and other members. But it'll still be breaking because methods that accept `Point` will no longer be able to accept `TileCoordinates`.

Reply:
> Wow, I wasn't expecting that much of a performance improvement just be not using generics! You've definitely got me there.

It's a very exaggerate case because additions is so cheap vs extra indirection of virtual function call. For shits and giggles, if you change my benchmark to actually return a new Point, i.e. implementing addition as non-mutating, the difference is 30+x